core/src/cubos/core/reflection/type.cpp

```diff
@@ -0,0 +1,71 @@
+#include <cubos/core/log.hpp>
+#include <cubos/core/reflection/type.hpp>
+
+using namespace cubos::core::reflection;
+
+Type& Type::create(std::string name)
+{
+    return *(new Type(name));
+}
+
+void Type::destroy(Type& type)
+{
+    delete &type;
+}
+
+const std::string& Type::name() const
+{
+    return mName;
+}
+
+Type& Type::with(uintptr_t id, void* trait, void (*deleter)(void*))
+{
+    CUBOS_ASSERT(!this->has(id), "Trait already present in type \"{}\"", this->name());
+
+    mTraits.emplace_back(Trait{
+        .id = id,
+        .value = trait,
+        .deleter = deleter,
+    });
+
+    return *this;
+}
+
+bool Type::has(uintptr_t id) const
+{
+    for (const auto& trait : mTraits)
+    {
+        if (trait.id == id)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+const void* Type::get(uintptr_t id) const
+{
+    for (const auto& trait : mTraits)
+    {
+        if (trait.id == id)
+        {
+            return trait.value;
+        }
+    }
+
+    CUBOS_FAIL("No such trait in type \"{}\"", this->name());
```

Comment:
Can we test asserts and aborts? Just to make codecov happy lmao

Reply:
We cant :pensive: doctest doesn't support it